app/src/main/java/com/nmakademija/nmaakademija/fragment/NewsFragment.java

```diff
@@ -102,4 +105,42 @@ public void onClick(View view) {
 
         }
     }
+
+    @Override
+    public void onArticlesUpdated(ArrayList<Article> articles) {
+        List<Article> oldAcademics = adapter.articlesList;
+        if (oldAcademics.size() == articles.size()) {
```

Comment:
Android support library has something for updating list. https://developer.android.com/reference/android/support/v7/util/DiffUtil.html
This will also add animations for inserted / removes items.